middleware-jackson/src/main/java/com/nytimes/android/external/store/middleware/jackson/JacksonParserFactory.java

```diff
@@ -26,8 +25,6 @@ private JacksonParserFactory() {
      */
     @Nonnull
     public static <T> Parser<String, T> createStringParser(@Nonnull JsonFactory jsonFactory, @Nonnull Type type) {
-        Preconditions.checkNotNull(jsonFactory, "jsonFactory cannot be null.");
```

Comment:
I'm OK dropping the dependency but would like to still have the checks here. Could you reimplement in line